### ISSUE TITLE
Two Agents on different servers connected to same Work Queue producing double results

### ISSUE THREAD

**Issue by robfreedy** [PrefectHQ/prefect#6772]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

We have gotten two Orion agents set up and running. They are both connected to the same work queue.  Unfortunately,  both agents pull the same jobs so everything is running on both servers and producing double results.  This is not our understanding of the purpose of the work queue.  For now we are going to limp along with 1 server but this is something we could really use help resolving.

### Reproduction

```python
Spinning up 2 agents on separate servers connecting to 1 work queue, then scheduling a flow and seeing that it. 

1. We are running the Prefect 2 agent in Docker on AWS Elastic Container Service (ECS).
     - The image is the prefecthq/prefect:2.3.1-python3.9 image from Docker Hub.
2. The agents are communicating with Prefect Cloud.
3. We are not customizing the agent profile except to provide the URL, API key, and work queue:
     - The URL is provided via the PREFECT_API_URL env var.
     - The API key is provided via the PREFECT_API_KEY env var.
     - The work queue is provided as an argument to the Docker entrypoint comment. We use: prefect agent start --work-queue default
```

### Error

No errors, just multiple flow runs when there is only supposed to be 1. 

### Versions

```
2.3.1

```

**Comment by robfreedy** [comment]
This issue will be resolved with this issue: https://github.com/PrefectHQ/prefect/issues/6725

**Comment by madkinsz** [comment]
This is surprising. The API should absolutely not allow two processes to enter a running state. Two instances of the infrastructure may start, but the flow run should not run twice.

**Comment by robfreedy** [comment]
Logs from the two agents as well: 
Agent 1: 
```
2022-09-08T13:08:05.763+00:00   13:08:05.763 | INFO | prefect.agent - Submitting flow run 'b662ec3e-55ad-4973-a182-670eac546de4'
    2022-09-08T13:08:06.133+00:00   13:08:06.133 | INFO | prefect.infrastructure.process - Opening process 'illustrious-wildebeest'...
    2022-09-08T13:08:06.138+00:00   13:08:06.137 | INFO | prefect.agent - Completed submission of flow run 'b662ec3e-55ad-4973-a182-670eac546de4'
    2022-09-08T13:08:09.102+00:00   13:08:09.101 | INFO | Flow run 'illustrious-wildebeest' - Suggesting privilege tags for upload with id: 6319e8b8456b1125261f7add
    2022-09-08T13:08:09.264+00:00   13:08:09.264 | INFO | Flow run 'illustrious-wildebeest' - Created task run 'prepare_processing_manifests-8ccac25d-0' for task 'prepare_processing_manifests'
    2022-09-08T13:08:09.265+00:00   13:08:09.264 | INFO | Flow run 'illustrious-wildebeest' - Executing 'prepare_processing_manifests-8ccac25d-0' immediately...
    2022-09-08T13:08:09.856+00:00   13:08:09.855 | INFO | Task run 'prepare_processing_manifests-8ccac25d-0' - Found 3 documents that need processing. Splitting into sub-manifests of 500 documents.
    2022-09-08T13:08:09.987+00:00   13:08:09.987 | INFO | Task run 'prepare_processing_manifests-8ccac25d-0' - Finished in state Completed()
    2022-09-08T13:08:10.127+00:00   13:08:10.126 | INFO | Flow run 'illustrious-wildebeest' - Created task run 'detect_privilege_in_manifests-5deb5d9d-0' for task 'detect_privilege_in_manifests'
    2022-09-08T13:08:10.127+00:00   13:08:10.127 | INFO | Flow run 'illustrious-wildebeest' - Executing 'detect_privilege_in_manifests-5deb5d9d-0' immediately...
    2022-09-08T13:08:10.350+00:00   13:08:10.350 | INFO | Task run 'detect_privilege_in_manifests-5deb5d9d-0' - Running concurrent privilege detection on 1 manifests
    2022-09-08T13:08:10.398+00:00   13:08:10.397 | INFO | Task run 'detect_privilege_in_manifests-5deb5d9d-0' - Running privilege detection for manifest 6319e8b8456b1125261f7add-0 with 3 entries
    2022-09-08T13:08:10.588+00:00   13:08:10.587 | INFO | Task run 'detect_privilege_in_manifests-5deb5d9d-0' - Privilege detection done for manifest 6319e8b8456b1125261f7add-0 with 3 entries
    2022-09-08T13:08:10.731+00:00   13:08:10.731 | INFO | Task run 'detect_privilege_in_manifests-5deb5d9d-0' - Finished in state Completed()
    2022-09-08T13:08:10.868+00:00   13:08:10.867 | INFO | Flow run 'illustrious-wildebeest' - Created task run 'convert_and_upload_results-aa969807-0' for task 'convert_and_upload_results'
    2022-09-08T13:08:10.868+00:00   13:08:10.867 | INFO | Flow run 'illustrious-wildebeest' - Executing 'convert_and_upload_results-aa969807-0' immediately...
    2022-09-08T13:08:11.409+00:00   13:08:11.409 | INFO | Task run 'convert_and_upload_results-aa969807-0' - Finished in state Completed()
    2022-09-08T13:08:11.548+00:00   13:08:11.547 | INFO | Flow run 'illustrious-wildebeest' - Created task run 'notify_results_available-0f54b487-0' for task 'notify_results_available'
    2022-09-08T13:08:11.548+00:00   13:08:11.547 | INFO | Flow run 'illustrious-wildebeest' - Executing 'notify_results_available-0f54b487-0' immediately...
    2022-09-08T13:08:11.932+00:00   13:08:11.932 | INFO | Task run 'notify_results_available-0f54b487-0' - Finished in state Completed()
    2022-09-08T13:08:12.045+00:00   13:08:12.045 | ERROR | Flow run 'illustrious-wildebeest' - Crash detected! Execution was interrupted by an unexpected exception.
    2022-09-08T13:08:12.169+00:00   13:08:12.169 | INFO | prefect.engine - Engine execution of flow run 'b662ec3e-55ad-4973-a182-670eac546de4' aborted by orchestrator: This run has already terminated.
    2022-09-08T13:08:12.621+00:00   13:08:12.621 | INFO | prefect.infrastructure.process - Process 'illustrious-wildebeest' exited cleanly.
```

Agent 2: 
```
 2022-09-08T13:08:05.833+00:00  13:08:05.833 | INFO | prefect.agent - Submitting flow run 'b662ec3e-55ad-4973-a182-670eac546de4'
    2022-09-08T13:08:06.191+00:00   13:08:06.191 | INFO | prefect.infrastructure.process - Opening process 'illustrious-wildebeest'...
    2022-09-08T13:08:06.196+00:00   13:08:06.195 | INFO | prefect.agent - Completed submission of flow run 'b662ec3e-55ad-4973-a182-670eac546de4'
    2022-09-08T13:08:09.232+00:00   13:08:09.231 | INFO | Flow run 'illustrious-wildebeest' - Suggesting privilege tags for upload with id: 6319e8b8456b1125261f7add
    2022-09-08T13:08:09.348+00:00   13:08:09.348 | INFO | Flow run 'illustrious-wildebeest' - Created task run 'prepare_processing_manifests-8ccac25d-0' for task 'prepare_processing_manifests'
    2022-09-08T13:08:09.349+00:00   13:08:09.348 | INFO | Flow run 'illustrious-wildebeest' - Executing 'prepare_processing_manifests-8ccac25d-0' immediately...
    2022-09-08T13:08:09.963+00:00   13:08:09.962 | INFO | Task run 'prepare_processing_manifests-8ccac25d-0' - Found 3 documents that need processing. Splitting into sub-manifests of 500 documents.
    2022-09-08T13:08:10.344+00:00   13:08:10.341 | ERROR | Flow run 'illustrious-wildebeest' - Encountered exception during execution:
    2022-09-08T13:08:10.344+00:00   Traceback (most recent call last):
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 1084, in begin_task_run
    2022-09-08T13:08:10.344+00:00   return await orchestrate_task_run(
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 1223, in orchestrate_task_run
    2022-09-08T13:08:10.344+00:00   state = await propose_state(
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 1456, in propose_state
    2022-09-08T13:08:10.344+00:00   raise prefect.exceptions.Abort(response.details.reason)
    2022-09-08T13:08:10.344+00:00   prefect.exceptions.Abort: This run has already terminated.
    2022-09-08T13:08:10.344+00:00   During handling of the above exception, another exception occurred:
    2022-09-08T13:08:10.344+00:00   Traceback (most recent call last):
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 595, in orchestrate_flow_run
    2022-09-08T13:08:10.344+00:00   result = await run_sync(flow_call)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 57, in run_sync_in_worker_thread
    2022-09-08T13:08:10.344+00:00   return await anyio.to_thread.run_sync(call, cancellable=True)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/anyio/to_thread.py", line 31, in run_sync
    2022-09-08T13:08:10.344+00:00   return await get_asynclib().run_sync_in_worker_thread(
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    2022-09-08T13:08:10.344+00:00   return await future
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    2022-09-08T13:08:10.344+00:00   result = context.run(func, *args)
    2022-09-08T13:08:10.344+00:00   File "insights/flows/suggest_privilege_tags.py", line 336, in suggest_privilege_tags
    2022-09-08T13:08:10.344+00:00   processing_manifests = prepare_processing_manifests(upload_id)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/tasks.py", line 295, in __call__
    2022-09-08T13:08:10.344+00:00   return enter_task_run_engine(
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 735, in enter_task_run_engine
    2022-09-08T13:08:10.344+00:00   return run_async_from_worker_thread(begin_run)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 137, in run_async_from_worker_thread
    2022-09-08T13:08:10.344+00:00   return anyio.from_thread.run(call)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/anyio/from_thread.py", line 49, in run
    2022-09-08T13:08:10.344+00:00   return asynclib.run_async_from_thread(func, *args)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 970, in run_async_from_thread
    2022-09-08T13:08:10.344+00:00   return f.result()
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 446, in result
    2022-09-08T13:08:10.344+00:00   return self.__get_result()
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    2022-09-08T13:08:10.344+00:00   raise self._exception
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 863, in get_task_call_return_value
    2022-09-08T13:08:10.344+00:00   return await future._result()
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/futures.py", line 236, in _result
    2022-09-08T13:08:10.344+00:00   return final_state.result(raise_on_failure=raise_on_failure)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/orion/schemas/states.py", line 143, in result
    2022-09-08T13:08:10.344+00:00   raise data
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/task_runners.py", line 203, in submit
    2022-09-08T13:08:10.344+00:00   result = await call()
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 1101, in begin_task_run
    2022-09-08T13:08:10.344+00:00   task_run.state.data._cache_data(await _retrieve_result(task_run.state))
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/results.py", line 38, in _retrieve_result
    2022-09-08T13:08:10.344+00:00   serialized_result = await _retrieve_serialized_result(state.data)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/client.py", line 103, in with_injected_client
    2022-09-08T13:08:10.344+00:00   return await fn(*args, **kwargs)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/results.py", line 34, in _retrieve_serialized_result
    2022-09-08T13:08:10.344+00:00   return await filesystem.read_path(result.key)
    2022-09-08T13:08:10.344+00:00   File "/usr/local/lib/python3.9/site-packages/prefect/filesystems.py", line 183, in read_path
    2022-09-08T13:08:10.344+00:00   raise ValueError(f"Path {path} does not exist.")
    2022-09-08T13:08:10.344+00:00   ValueError: Path /root/.prefect/storage/a4dec93f0b3840f381f66d868875fa71 does not exist.
    2022-09-08T13:08:10.344+00:00   13:08:10.344 | INFO | Task run 'prepare_processing_manifests-8ccac25d-0' - Crash detected! Execution was interrupted by an unexpected exception.
    2022-09-08T13:08:10.592+00:00   13:08:10.588 | ERROR | Flow run 'illustrious-wildebeest' - Finished in state Failed('Flow run encountered an exception.')
    2022-09-08T13:08:11.143+00:00   13:08:11.143 | INFO | prefect.infrastructure.process - Process 'illustrious-wildebeest' exited cleanly.
    2022-09-08T13:11:42.646+00:00   13:11:42.646 | INFO | prefect.agent - Submitting flow run '28a40210-ad72-47a9-a43c-8a8303d50edc'
```

**Comment by knl** [comment]
I'm observing the same issue. In my case, there are 4 agents and one orion server running on the same machine, using the same conda environments. I observe frequently (but not every time) that there are duplicated subflow or flow runs.

**Comment by madkinsz** [comment]
What version is your server? I believe we have resolved this with #6852

**Comment by madkinsz** [comment]
Sorry that's unreleased! It'll be out today and should close this issue :)

**Comment by knl** [comment]
> Sorry that's unreleased! It'll be out today and should close this issue :)

Awesome, will try it as soon as the new version is rolled out.

I'm running the latest, btw:

```
Version:             2.4.0
API version:         0.8.0
Python version:      3.7.9
Git commit:          513639e8
Built:               Tue, Sep 13, 2022 2:15 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         hosted
```

**Comment by jlowin** [comment]
I believe this was fixed in #6852 which has been released
